RabbitMQ.Stream.Client/Connection.cs

```diff
@@ -85,41 +86,28 @@ private static System.IO.Stream MaybeTcpUpgrade(NetworkStream networkStream, Ssl
             return new Connection(socket, commandCallback, closedCallBack, sslOption);
         }
 
-        private ValueTask<FlushResult> FlushAsync()
-        {
-            lock (writeLock)
-            {
-                var flushTask = writer.FlushAsync();
-                return flushTask;
-            }
-        }
-
         public async ValueTask<bool> Write<T>(T command) where T : struct, ICommand
         {
-            WriteCommand(command);
-            var flushTask = FlushAsync();
-
-            // Let's check if this is completed synchronously before invoking the async state machine
-            if (!flushTask.IsCompletedSuccessfully)
-            {
-                await flushTask.ConfigureAwait(false);
-            }
-
-            return flushTask.Result.IsCompleted;
+            await WriteCommand(command);
```

Comment:
Now everything is async with the `await` we `return true;` to leave the API compatibility